rust/src/lib.rs

```diff
@@ -184,24 +186,85 @@ pub fn add_to_linker<T: 'static>(
 ) -> Result<EventChannel> {
     let (ctx, events_tx, events_rx) = InstrumentationContext::new(id);
 
-    // Build up a table of function id to name mapping
+    // Parse the wasm to build up a table of function id to name mapping and check version compat
     let mut function_names = FunctionNames::new();
+    let mut maj_index: Option<u32> = None;
+    let mut min_index: Option<u32> = None;
+    let mut globals = HashMap::<u32, u32>::new();
     let parser = wasmparser::Parser::new(0);
     for payload in parser.parse_all(data) {
-        if let wasmparser::Payload::CustomSection(custom) = payload? {
-            if custom.name() == "name" {
-                let name_reader =
-                    wasmparser::NameSectionReader::new(custom.data(), custom.data_offset());
-                for x in name_reader.into_iter() {
-                    if let wasmparser::Name::Function(f) = x? {
-                        for k in f.into_iter() {
-                            let k = k?;
-                            function_names.insert(k.index, k.name.to_string());
+        match payload? {
+            wasmparser::Payload::CustomSection(custom) => {
+                if custom.name() == "name" {
+                    let name_reader =
+                        wasmparser::NameSectionReader::new(custom.data(), custom.data_offset());
+                    for x in name_reader.into_iter() {
+                        if let wasmparser::Name::Function(f) = x? {
+                            for k in f.into_iter() {
+                                let k = k?;
+                                function_names.insert(k.index, k.name.to_string());
+                            }
                         }
                     }
+                    continue;
                 }
-                continue;
             }
+            wasmparser::Payload::GlobalSection(globalsec) => {
```

Comment:
This seems good but just wanted to mention that if the module has already been registered with `linker` we could probably just get the global value that way. But maybe that isn't a valid assumption to make.

Reply:
We probably could, but having it runtime independent could be helpful.